custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -780,25 +784,80 @@ def _prepare_prediction_server_or_batch_pipeline(self, run_language):
 
         return DrumUtils.render_file(functional_pipeline_filepath, replace_data)
 
-    def _run_predictions_pipelines_in_mlpiper(self):
-        run_language = self._check_artifacts_and_get_run_language()
+    def _run_predictions(self):
+        if self.run_mode != RunMode.SCORE:
+            raise NotImplemented("Only score mode is supported here")
 
-        if self.run_mode == RunMode.SERVER:
-            # in prediction server mode infra pipeline == prediction server runner pipeline
-            infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
-        elif self.run_mode == RunMode.SCORE:
-            tmp_output_filename = None
-            # if output is not provided, output into tmp file and print
-            if not self.options.output:
-                # keep object reference so it will be destroyed only in the end of the process
-                __tmp_output_file = tempfile.NamedTemporaryFile(mode="w")
-                self.options.output = tmp_output_filename = __tmp_output_file.name
-            # in batch prediction mode infra pipeline == predictor pipeline
+        with self._setup_output_if_not_exists():
```

Comment:
Can you explain why you're doing options output file setup as a contextmanager instead of directly calling `self._setup_output_if_not_exists()` without yields?

Reply:
Because the printout should take place only after the custom model was executed. This is a more elegant implementation of the previous code.